pex/pip/tool.py

```diff
@@ -451,7 +452,7 @@ def spawn_download_distributions(
 
         if not build_isolation:
             download_cmd.append("--no-build-isolation")
-            extra_env.update(PEP517_BACKEND_PATH=os.pathsep.join(sys.path))
+            pex_extra_sys_path.extend(sys.path)
```

Comment:
Now that Pex uses an in-tree build backend, it unearthed this latent bug where `PEP517_BACKEND_PATH` would get stomped with the backend-path. As such, both here in spawn download and lower in spawn install wheel, we now pass non-isolated-build sys.path via PEX_EXTRA_SYS_PATH instead.